### PR TITLE
DAOS-10608 test: Run vfio-pci workaround fix between tests

### PR DESCRIPTION
**Pull request by phender**
Adding running the vfio-pci binding fix workaround commands between
running avocado test files on the hosts that were used by the test to
run the daos_server command.

Also update the stopping of server and clients to limit commands to
hosts only used by the test to run servers and clients, respectively.

Skip-unit-tests: true
Test-tag-hw-large: pr full_regression

Signed-off-by: Phillip Henderson <phillip.henderson@intel.com>

### CODE REVIEW

**Review by tanabarr — DISMISSED** [review]

File: src/tests/ftest/launch.py [hunk 1]
```diff
@@ -2098,41 +2099,43 @@ def stop_daos_agent_services(test_file, args):
...
-        hosts = list(args.test_clients)
-    else:
-        hosts = list(args.test_servers)
+    client_hosts = get_hosts_from_yaml(test_file["yaml"], args, YAML_KEYS["test_clients"])
     local_host = socket.gethostname().split(".")[0]
-    if local_host not in hosts:
-        hosts.append(local_host)
-    return stop_service(hosts, service)
+    if local_host not in client_hosts:
+        client_hosts.append(local_host)
```
Comment (daltonbohning) [inline src/tests/ftest/launch.py:2113]: Would it make sense for `get_hosts_from_yaml` to return a `NodeSet`, similar to the definition in `test.py`?

File: src/tests/ftest/launch.py [hunk 2]
```diff
@@ -2240,6 +2243,31 @@ def indent_text(indent, text):
...
+    This is a workaround to enable binding devices back to nvme or vfio-pci after they are unbound
+    from vfio-pci to nvme.  This should resolve the "NVMe not found" error seen when attempting to
+    start daos engines in the test.
+
+    Args:
+        test_file (dict): a dictionary of the test script/yaml file
+        args (argparse.Namespace): command line arguments for this program
+
+    Returns:
+        int: status code: 0 = success, 512 = failure
```
Comment (daltonbohning) [inline src/tests/ftest/launch.py:2258]: Looks like the only return is `return 0`

Comment (phender) [reply src/tests/ftest/launch.py:2258]: That's intended.  If we ever expand this function to return a failure it should return 512, hence the comment.

File: src/tests/ftest/launch.py (hunk 1 above)
Comment (phender) [reply src/tests/ftest/launch.py:2113]: That's being addressed in another PR and a bit beyond the scope of this change.

**Review by daltonbohning — APPROVED** [review]

**Comment by phender** [comment]
Sample output on VMs:
```
[2022-06-09T16:52:30.170Z] --------------------------------------------------------------------------------
[2022-06-09T16:52:30.170Z] Resetting server storage on wolf-105vm[2-3] after running './cart/iv_two_node.py'
[2022-06-09T16:52:30.170Z] Running on wolf-105vm[2-3]: bash -c 'if lspci | grep -i nvme;then daos_server storage prepare -n --reset && sudo rmmod vfio_pci && sudo modprobe vfio_pci;fi'
[2022-06-09T16:52:30.170Z]     wolf-105vm[2-3]: rc=0, output: <NONE>
[2022-06-09T16:52:30.170Z] --------------------------------------------------------------------------------
```
and HW clusters:
```
[2022-06-09T20:06:45.804Z] --------------------------------------------------------------------------------
[2022-06-09T20:06:45.804Z] Resetting server storage on wolf-[201-207] after running './pool/create_capacity.py'
[2022-06-09T20:06:45.804Z] Running on wolf-[201-207]: bash -c 'if lspci | grep -i nvme;then daos_server storage prepare -n --reset && sudo rmmod vfio_pci && sudo modprobe vfio_pci;fi'
[2022-06-09T20:06:45.804Z]     wolf-[201-207]: rc=0, output:
[2022-06-09T20:06:45.804Z]       5e:00.0 Non-Volatile memory controller: Intel Corporation NVMe Datacenter SSD [3DNAND, Beta Rock Controller]
[2022-06-09T20:06:45.804Z]       5f:00.0 Non-Volatile memory controller: Intel Corporation NVMe Datacenter SSD [3DNAND, Beta Rock Controller]
[2022-06-09T20:06:45.804Z]       81:00.0 Non-Volatile memory controller: Intel Corporation NVMe Datacenter SSD [Optane]
[2022-06-09T20:06:45.804Z]       86:00.0 Non-Volatile memory controller: Intel Corporation NVMe Datacenter SSD [Optane]
[2022-06-09T20:06:45.804Z]       Reset locally-attached NVMe storage...
[2022-06-09T20:06:45.804Z] --------------------------------------------------------------------------------
```

**Review by tanabarr — APPROVED** [review]

**Review by daltonbohning — APPROVED** [review]

**Pull request merged**